network/benchmarks/netperf/lib/outputlib.go

```diff
@@ -0,0 +1,68 @@
+package lib
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	api "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func getLogsFromPod(c *kubernetes.Clientset, podName, testNamespace string) (*string, error) {
+	var logData *string
+	backoff := 10 * time.Second
+
+	for i := 0; i < 3; i++ {
+		body, err := c.CoreV1().Pods(testNamespace).GetLogs(podName, &api.PodLogOptions{}).DoRaw(context.Background())
+		if err == nil {
+			data := string(body)
+			logData = &data
+			return logData, nil
+		}
+		time.Sleep(backoff)
+		backoff *= 2
+	}
+
+	return nil, fmt.Errorf("error reading logs from pod %s after 3 retries", podName)
+}
+
+func getDataFromPod(c *kubernetes.Clientset, podName, startMarker, endMarker, testNamespace string) (*string, error) {
+	logData, err := getLogsFromPod(c, podName, testNamespace)
+	if err != nil {
+		return nil, err
+	}
+	index := strings.Index(*logData, startMarker)
+	endIndex := strings.Index(*logData, endMarker)
+	if index == -1 || endIndex == -1 {
+		return nil, nil
```

Comment:
is this a valid case?

Reply:
We basically poll the logs to check if the results have been published, the program exits when it recieves either a result or an error, otherwise it polls again.